### PR TITLE
Add new chunked_string type

### PR DESCRIPTION
**Pull request by Neverlord**
This is prep work for #1652.

The idea here is to have log events use the monotonic buffer resource to build up the message and (optional) fields. However, we don't want `std::string` or `std::vector<char>` inside of a buffer resource, because it would produce gaps (by re-allocating the internal buffer when pushing to it). The chunked strings allow us to store the result from `detail::format` (or `std::format`) without wasting memory blocks and it also avoids unnecessary copies when growing the buffer. The chunked strings are ideal for generating log events, because we build up a message and/or field values and then consume the whole thing once (for generating file and/or console output) before releasing the memory resources again. The chunked string is a bit more inconvenient to use then "regular" strings, but that'll be well worth it.

This PR only adds the new type `caf::linked_string_chunk`. Integrating it into the logger will be a followup.

### CODE REVIEW

File: libcaf_core/caf/linked_string_chunk.hpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    return *this;
+  }
+
+  linked_string_chunk_builder_output_iterator& operator++() {
+    return *this;
+  }
+
+  linked_string_chunk_builder_output_iterator operator++(int) {
+    return *this;
+  }
```
Comment (riemass) [inline libcaf_core/caf/linked_string_chunk.hpp:106]: First time reading the implementation of a pure output iterator. It's a bit wired, but should be correct, at least from what I see on cppreference.

**Review by riemass — APPROVED** [review]
The idea is great and we have something similar with our `json` implementation.

Initially reviewing this, I think the class names and use cases are hard to comprehend.
Regarding class names, I would really like to see the change from `linked_string_chunk` to `linked_string_node`, since that is a bit more common name. 

The use cases are hard to comprehend because It's hard to reason what function operates on the whole linked list, and when only on the pointed to chunk. For example reading the function signature:
```
void render_val(const linked_string_chunk* val) {
```
knowing that the data structure is actually a single link node, and seeing that we have a `const*` I would assume this function renders the whole list.
And when reading the signature:
```
std::string to_string(const linked_string_chunk& head) {
```
since I see `const&` I would assume only one chunk gets converted to the string, however, here again we iterate till the end and convert everything.

I would like to, at least, change the signatures to consistently use `const*` and leave our references. Sadly, it would be hard having a `linked_string_list` type and be clear about copy semantics (or the lack of). If the API will be used extensively in our new logger, it might be worth giving it a taught. 

Since this is more of a discussion item, I still approve the PR.

**Comment by Neverlord** [comment]
Thanks for the feedback! I've re-organized the thing a bit. There's not `chunked_string`, which is what the user usually would care about. This type has the member functions that operate on the whole thing now. The linked list underneath uses `chunked_string::node`. I think this makes the API a lot easier to navigate now. Thoughts?

**Comment by riemass** [comment]
The structure looks great now. We will get used to the shallow copy semantics.

**Review by riemass — APPROVED** [review]
Gave it another look. This is

**Pull request merged**